daemon/restart.go

```diff
@@ -39,8 +39,10 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		return err
 	}
 
-	if err := daemon.containerStart(container); err != nil {
-		return err
+	if !container.HostConfig.AutoRemove {
```

Comment:
It's also restricted by technique, I can't find a method to distinguish stop command from `docker restart` or pure `docker stop`, I can only detect the container is stopped and should be removed.